zeta/templates/zeta.template.c

```diff
@@ -87,6 +100,24 @@ int zt_chan_read(zt_channel_e id, zt_data_t *channel_data)
         ZT_CHECK(k_sem_take(channel->sem, K_MSEC(200)) != 0, -EBUSY,
                  "Could not read the channel. Channel is busy");
         memcpy(channel_data->bytes.value, channel->data, channel->size);
+
+#ifdef CONFIG_ZETA_FORWARDER
+        zt_service_t *current_service = NULL;
+        for (int i = 0; i < ZT_SERVICE_COUNT; ++i) {
+            if ((*__zt_services[i]->thread_id) == k_current_get()) {
+                current_service = __zt_services[i];
+                break;
+            }
+        }
```

Comment:
Too inefficient. It would be better to access the current service by using an enum with the position of the services.  We can do this generating the enum based on the position of the services into the services array. Did you get it?

Reply:
I thought about that but there are some problems. The only thing that recognizes what service is running is the k_current_get() that returns const k_tid_t *. This cannot be used as a position in an array. I tried to look for something inside this struct that may be used but I couldn't find out. Do you have other suggestions or some approach to convert the k_current_get() return in something that may be used as an enum that is indexing a service array?

Reply:
We changed the approach of accessing zephyr thread by zt_service_t and now we are using CONTAINER_OF to solve the problem of performance.